pkg/operator/ceph/object/dependents.go

```diff
@@ -106,3 +127,39 @@ func getBucketDependents(
 
 	return nil
 }
+
+func getMasterZoneDependents(
+	deps *dependents.DependentList,
+	clusterdCtx *clusterd.Context,
+	clusterInfo *client.ClusterInfo,
+	store *v1.CephObjectStore,
+	objContext *Context,
+) error {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+	zoneGroupOutput, err := RunAdminCommandNoMultisite(objContext, true, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch zonegroup %q", objContext.ZoneGroup)
+	}
+	zoneGroupJson, err := DecodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return errors.Wrapf(err, "failed to decode zonegroup %q", objContext.ZoneGroup)
+	}
+	// if only master zone remains then it is okay delete
+	if len(zoneGroupJson.Zones) > 1 {
+		peerZones := "["
+		for _, zoneList := range zoneGroupJson.Zones {
```

Comment:
Each element of zones is not a list but a zone.

```suggestion
		for _, zone := range zoneGroupJson.Zones {
```